backend/src/src-fossology/src/main/java/com/siemens/sw360/fossology/handler/FossologyFileHandler.java

```diff
@@ -191,6 +192,28 @@ public Release getStatusInFossology(String releaseId, User user, String clearing
         return release;
     }
 
+    private void updateReleaseClearingState(Release release, FossologyStatus fossologyStatus) {
+        updateReleaseClearingState(release, Optional.of(fossologyStatus));
+    }
+    private void updateReleaseClearingState(Release release, Optional<FossologyStatus> fossologyStatus) {
+        Optional<ClearingState> newClearingState = mapFossologyStatusToClearingState(fossologyStatus);
```

Comment:
if you replace this with a call of the form
```
    Optional<ClearingState> newClearingState = fossologyStatus.flatMap(...)
```
the function `mapFossologyStatusToClearingState` would become
```
    private Optional<ClearingState> mapFossologyStatusToClearingState(FossologyStatus fossologyStatus) {
```
and you do not have to extract the content of the optional manually.